src/main/java/com/google/devtools/build/lib/exec/SpawnRunner.java

```diff
@@ -198,6 +199,14 @@ default ArtifactPathResolver getPathResolver() {
 
     /** Reports a progress update to the Spawn strategy. */
     void report(ProgressStatus state, String name);
+
+    /**
+     * Returns a {@link MetadataInjector} that allows a caller to inject metadata about spawn
+     * outputs that are stored remotely.
+     */
+    default MetadataInjector getMetadataInjector() {
```

Comment:
AFAIU this will let and SpawnRunner inject metadata for artifacts that is not on its inputs, which could result in bugs that are hard to track down. Is this case caught somewhere I didn't notice? If not, can we do that?

Reply:
done